app/src/main/java/com/github/gotify/messages/MessagesActivity.java

```diff
@@ -568,9 +575,59 @@ public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == R.id.action_delete_all) {
             new DeleteMessages().execute(appId);
         }
+        if (item.getItemId() == R.id.action_delete_app) {
+            android.app.AlertDialog.Builder alert = new android.app.AlertDialog.Builder(this);
+            alert.setTitle(R.string.delete_app);
+            alert.setMessage("Are you sure?");
+            alert.setPositiveButton(
+                    "YES",
+                    new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            deleteApp(appId);
+                        }
+                    });
+            alert.setNegativeButton(
+                    "NO",
+                    new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {}
+                    });
+            alert.show();
+        }
         return super.onContextItemSelected(item);
     }
 
+    private void deleteApp(Long appId) {
+        Thread thread =
```

Comment:
You can use the async version of the api call instead of creating a thread for this:
```java
private void deleteApp(Long appId) {
    ApiClient client =
            ClientFactory.clientToken(settings.url(), settings.sslSettings(), settings.token());

    client.createService(ApplicationApi.class)
            .deleteApp(appId)
            .enqueue(
                    Callback.callInUI(
                            this,
                            (ignored) -> refreshAll(),
                            (e) -> Utils.showSnackBar(this, "Could not delete app")));
}
```
"Could not delete app" should be put into strings.xml